test/plog/LogStreamTest.cpp

```diff
@@ -0,0 +1,449 @@
+            })
+            .handle_exception([this](auto exc) {
+                try {
+                    std::rethrow_exception(exc);
+                } catch (RPCDispatcher::RequestTimeoutException& exc) {
+                    K2LOG_E(log::ltest, "======= Test failed due to timeout ========");
+                    exitcode = -1;
+                } catch (std::exception& e) {
+                    K2LOG_E(log::ltest, "======= Test failed with exception [{}] ========", e.what());
+                    exitcode = -1;
+                }
+            })
+            .finally([this] {
+                K2LOG_I(log::ltest, "======= Test ended ========");
+                seastar::engine().exit(exitcode);
+            });
+        });
+        _testTimer.arm(0ms);
+        return seastar::make_ready_future<>();
+    }
+
+    seastar::future<> runTest1() {
+        K2LOG_I(log::ltest, ">>> Test1: Write and Read a Value from/to one log stream");
+        _logStream = _mmgr.obtainLogStream(LogStreamType::WAL);
+        String header("0", 10000);
+        Payload payload([] { return Binary(20000); });
+        payload.write(header);
+        // write a string to the first log stream
+        return _logStream->append(std::move(payload))
+        .then([this, header] (auto&& response){
+            auto& [status, plogId, appended_offset] = response;
+            if (!status.is2xxOK()){
+                throw std::runtime_error("cannot append data!");
+            }
+            _initPlogId = plogId;
+            return _logStream->read(plogId, 0, appended_offset);
+        })
+        .then([this, header] (auto&& response){
+            auto& [status, continuation_token, payload] = response;
+            if (!status.is2xxOK()){
+                throw std::runtime_error("cannot read data!");
+            }
+            K2EXPECT(log::ltest, payload.getSize(), 10005);
+            String str;
+            payload.seek(0);
+            payload.read(str);
+            K2EXPECT(log::ltest, str, header);
+            return seastar::make_ready_future<>();
+        });
+    }
+
+    seastar::future<> runTest2() {
+        K2LOG_I(log::ltest, ">>> Test2: Write and read huge data");
+         _logStream = _mmgr.obtainLogStream(LogStreamType::IndexerSnapshot);
+        // write 1000 Strings to the second log stream, these writes will trigger a plog switch
+        String header("0", 10000);
+        std::vector<seastar::future<std::tuple<Status, String, uint32_t>> > writeFutures;
+        for (uint32_t i = 0; i < 2000; ++i){
+            Payload payload([] { return Binary(20000); });
+            payload.write(header);
+            writeFutures.push_back(_logStream->append(std::move(payload)));
+        }
+        // read all the Strings that we wrote before, check whether they are the same
+        return seastar::when_all_succeed(writeFutures.begin(), writeFutures.end())
+        .then([this, header] (std::vector<std::tuple<Status, String, uint32_t> >&& responses){
```

Comment:
why capture header here?

Reply:
The header is a little bit confused. I have changed it to data_to_append